src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java

```diff
@@ -75,6 +88,231 @@ public WorkflowStepFactory(
+            new TimeValue(15, SECONDS)
+        ),
+
+        UNDEPLOY_MODEL("undeploy_model", Arrays.asList("model_id"), Arrays.asList("success"), Arrays.asList("opensearch-ml"), null),
+
+        DELETE_MODEL("delete_model", Arrays.asList("model_id"), Arrays.asList("model_id"), Arrays.asList("opensearch-ml"), null),
+
+        DELETE_CONNECTOR(
+            "delete_connector",
+            Arrays.asList("connector_id"),
+            Arrays.asList("connector_id"),
+            Arrays.asList("opensearch-ml"),
+            null
+        ),
+
+        REGISTER_AGENT("register_agent", Arrays.asList("name", "type"), Arrays.asList("agent_id"), Arrays.asList("opensearch-ml"), null),
+
+        DELETE_AGENT("delete_agent", Arrays.asList("agent_id"), Arrays.asList("agent_id"), Arrays.asList("opensearch-ml"), null),
+
+        CREATE_TOOL("create_tool", Arrays.asList("type"), Arrays.asList("tools"), Arrays.asList("opensearch-ml"), null);
+
+        private final String workflowStep;
+        private final List<String> inputs;
+        private final List<String> outputs;
+        private final List<String> requiredPlugins;
+        private final TimeValue timeout;
+
+        private static final List<String> allWorkflowSteps = Stream.of(values())
+            .map(WorkflowSteps::getWorkflowStep)
+            .collect(Collectors.toList());
+
+        WorkflowSteps(String workflowStep, List<String> inputs, List<String> outputs, List<String> requiredPlugins, TimeValue timeout) {
+            this.workflowStep = workflowStep;
+            this.inputs = List.copyOf(inputs);
+            this.outputs = List.copyOf(outputs);
+            this.requiredPlugins = requiredPlugins;
+            this.timeout = timeout;
+        }
+
+        /**
+         * Returns the workflowStep for the given enum Constant
+         * @return the workflowStep of this data.
+         */
+        public String getWorkflowStep() {
+            return workflowStep;
+        }
+
+        public List<String> getInputs() {
+            return inputs;
+        }
+
+        public List<String> getOutputs() {
+            return outputs;
+        }
+
+        public List<String> getRequiredPlugins() {
+            return requiredPlugins;
+        }
+
+        public TimeValue getTimeout() {
+            return timeout;
+        }
+
+        public WorkflowStepValidator getWorkflowStepValidator() {
+            return new WorkflowStepValidator(workflowStep, inputs, outputs, requiredPlugins, timeout);
+        };
+
+        /**
+         * Gets the timeout based on the workflowStep.
+         * @param workflowStep workflow step type
+         * @return the resource that will be created
+         * @throws FlowFrameworkException if workflow step doesn't exist in enum
+         */
+        public static TimeValue getTimeoutByWorkflowType(String workflowStep) throws FlowFrameworkException {
+            if (workflowStep != null && !workflowStep.isEmpty()) {
```

Comment:
o.o.core.common.Strings has you covered here:
```suggestion
            if (!Strings.isNullOrEmpty(workflowStep) {
```